feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt

```diff
@@ -137,21 +145,44 @@ sealed class StakeViewState<S>(
         }
     }
 
+    @ExperimentalCoroutinesApi
+    fun countdownTimerFlow(millis: Long) = callbackFlow<Long> {
+        val timer = object : CountDownTimer(millis, 1000) {
+            override fun onTick(millisUntilFinished: Long) {
+                offer(millisUntilFinished)
+            }
+
+            override fun onFinish() {
+                offer(0)
+
+                close()
+            }
+        }
+
+        timer.start()
+
+        awaitClose { timer.cancel() }
+    }.flowOn(Dispatchers.Main)
+
+    @ExperimentalCoroutinesApi
     private suspend fun summaryFlow(): Flow<StakeSummaryModel<S>> {
         return combine(
             summaryFlowProvider(stakeState),
-            currentAssetFlow
-        ) { summary, asset ->
+            currentAssetFlow,
+            countdownTimerFlow(stakingInteractor.getTimeLeft().toLong())
+        ) { summary, asset, timer ->
             val token = asset.token
             val tokenType = token.type
 
+            val timeLeftString = if (summary.status is NominatorStatus.Waiting) timer.formatTime() else null
```

Comment:
Since `timeLeft` only makes sense in the context of Waiting status I suggest putting it as a field to `NominatorStatus.Waiting` and filling it in `statusResolver`. With such an approach there will be no nullable field and `timeLeft` existence will be guaranteed at compile time.

Reply:
Done